src/refract/JSONSchemaVisitor.cc

```diff
@@ -340,62 +336,22 @@ namespace refract
 
     void JSONSchemaVisitor::visit(const NullElement& e)
     {
-        NullElement nil;
-        addMember("type", &nil);
+        addMember("type", new NullElement);
     }
 
     void JSONSchemaVisitor::visit(const StringElement& e)
     {
-        const StringElement::ValueType* v = GetValue<StringElement>(e);
-
-        if (v) {
-            setSchemaType("string");
-
-            if (fixed && !e.empty()) {
-                ArrayElement *a = new ArrayElement;
-                StringElement *s = new StringElement;
-
-                s->set(*v);
-                a->push_back(s);
-                addMember("enum", a);
-            }
-        }
+        primitiveType(e, GetValue<StringElement>(e));
```

Comment:
I was getting an error something like `expected primary-expression before ‘>’ token` which I ahve to admit I don't understand .... need to dive deeper into templates :)